### PR TITLE
EVG-5464 Terminate hosts when removing static distros

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/host/static_hosts.go [hunk 1]
```diff
@@ -20,14 +20,9 @@ import (
 // If the distro is the empty string ("") then this operation affects
 // all distros.
 func MarkInactiveStaticHosts(activeStaticHosts []string, distroID string) error {
-	if len(activeStaticHosts) == 0 {
```
Comment (ybrill) [inline model/host/static_hosts.go]: in scheduler/wrapper.go:200 we still check this and skip the update

File: rest/data/distro.go [hunk 2]
```diff
@@ -69,13 +70,19 @@ func (dc *DBDistroConnector) CreateDistro(distro *distro.Distro) error {
...
-	err := distro.Remove(distroId)
-	if err != nil {
+	var err error
+	if err = distro.Remove(distroId); err != nil {
 		return gimlet.ErrorResponse{
 			StatusCode: http.StatusInternalServerError,
 			Message:    fmt.Sprintf("distro with id '%s' was not deleted", distroId),
 		}
 	}
+	if err = host.MarkInactiveStaticHosts([]string{}, distroId); err != nil {
```
Comment (ybrill) [inline rest/data/distro.go]: If the host cleanup fails do we want the distro to be removed? 
In other words, do we want to remove the hosts and then remove the distro so we're sure there are no orphaned hosts?

File: model/host/static_hosts.go (hunk 1 above)
Comment (bsamek) [reply model/host/static_hosts.go]: Fixed.

File: rest/data/distro.go (hunk 2 above)
Comment (bsamek) [reply rest/data/distro.go]: I inverted the cases.

**Comment by ybrill** [comment]
Looks good to me.

**Pull request merged**